src/components/plugin/glossary-popup.tsx

```diff
@@ -24,21 +39,41 @@ interface IProps {
   videoCaption?: string;
   secondLanguage?: string;
   onLanguageChange?: () => void;
+  studentInfo?: IStudentInfo;
+  demoMode?: boolean;
 }
 
 interface IState {
   currentUserDefinition: string;
   questionVisible: boolean;
+  canRecord: boolean;
+  recordingState: RecordingState;
+  recordingStartTime: number;
+  errorSavingRecording: boolean;
 }
 
 export default class GlossaryPopup extends React.Component<IProps, IState> {
   public static contextType = pluginContext;
 
   public state: IState = {
     currentUserDefinition: "",
-    questionVisible: this.props.askForUserDefinition || false
+    questionVisible: this.props.askForUserDefinition || false,
+    canRecord: this.canRecord(this.props),
+    recordingState: RecordingState.NotRecording,
+    recordingStartTime: 0,
+    errorSavingRecording: false
   };
 
+  private mediaRecorder: MediaRecorder | null = null;
+  private recordedBlobs: Blob[] = [];
+  private audio: HTMLAudioElement | null = null;
+  private recordingTimeout: number;
+
+  public UNSAFE_componentWillReceiveProps(nextProps: IProps) {
+    // teacher may change the recording option in their dashboard while popup is visible for student
+    this.setState({canRecord: this.canRecord(nextProps)});
```

Comment:
Not a big deal, but it seems `this.caneRecord` is a simple function that derives its output from props. It doesn't seem necessary to keep that value in the state I think. I can't see any place where we change this value after initialization. So, it `canRecord` could be a getter that we use in various places and then we could also avoid this UNSAFE_componentWillReceiveProps.